zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/distribution/DbDistributionState.java

```diff
@@ -55,6 +55,12 @@ public class DbDistributionState implements MutableDistributionState {
   private final DbCompositeKey<DbString, DbCompositeKey<DbInt, DbForeignKey<DbLong>>>
       queuedDistributionKey;
 
+  /** [queue id | key] => persisted continuation command */
+  private final ColumnFamily<DbCompositeKey<DbString, DbLong>, PersistedCommandDistribution>
+      continuationCommandColumnFamily;
```

Comment:
🔧 By using the queue id and distributionKey as the key of this CF you add a limitation that we can only add 1 continuation command per distribution. Is this intended? In the PR description you mention:

> In particular, we allow registering an arbitrary amount of commands that will be written once a queue is or becomes empty.

This is only true across different distributions now.

Reply:
The key is not a distribution key, it can be user provided. That allows registering multiple continuations on the same queue.

Reply:
Okay, so we'll have to generate a new key fo reach continuation. That sounds good to me. The naming in the builder confused me here. Could we rename this?

```java
    @Override
    public <T extends UnifiedRecordValue> void continueWith(final TypedRecord<T> command) {
      requestContinuation(
          queue,
          distributionKey, <--------
          awaitNonEmptyQueue,
          command.getValueType(),
          command.getIntent(),
          command.getValue());
    }

    @Override
    public <T extends UnifiedRecordValue> void continueWith(
        final ValueType valueType, final Intent intent, final T value) {
      requestContinuation(queue, distributionKey, <------ awaitNonEmptyQueue, valueType, intent, value);
    }
```

Reply:
Ah yeah my bad, this is leftover from refactoring :+1:

Reply:
Fixed in f389974ef272f99f3016feff097ed9eea12ce706